tools/runners/run-test-suite.py

```diff
@@ -0,0 +1,214 @@
+    if sys.platform == 'win32':
+        return ['cmd', '/S', '/C']
+    return []
+
+
+def execute_test_command(test_cmd):
+    process = subprocess.Popen(test_cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
+    stdout = process.communicate()[0]
+    return (process.returncode, stdout)
+
+
+def main(args):
+    tests = get_tests(args.test_dir, args.test_list, args.skip_list)
+    total = len(tests)
+    if total == 0:
+        print("No test to execute.")
+        return 1
+
+    if sys.platform == 'win32':
+        original_timezone = util.get_timezone()
+        util.set_sighdl_to_reset_timezone(original_timezone)
+        util.set_timezone('UTC')
+
+    if args.snapshot:
+        passed = run_snapshot_tests(args, tests)
+    else:
+        passed = run_normal_tests(args, tests)
+
+    if sys.platform == 'win32':
+        util.set_timezone(original_timezone)
+
+    failed = total - passed
+
+    summary_list = [os.path.relpath(args.engine)]
+    if args.snapshot:
+        summary_list.append('--snapshot')
+    if args.test_dir:
+        summary_list.append(os.path.relpath(args.test_dir))
+    if args.test_list:
+        summary_list.append(os.path.relpath(args.test_list))
+    util.print_test_summary(' '.join(summary_list), total, passed, failed)
+
+    if failed > 0:
+        return 1
+    return 0
+
+
+def run_normal_tests(args, tests):
+    test_cmd = get_platform_cmd_prefix()
+    if args.runtime:
+        test_cmd.append(args.runtime)
+    test_cmd.append(args.engine)
+
+    total = len(tests)
+    tested = 0
+    passed = 0
+    for test in tests:
+        tested += 1
+        test_path = os.path.relpath(test)
+        is_expected_to_fail = os.path.join(os.path.sep, 'fail', '') in test
+        (returncode, stdout) = execute_test_command(test_cmd + [test])
+
+        if bool(returncode) == is_expected_to_fail:
+            passed += 1
+            if not args.quiet:
+                passed_string = 'PASS' + (' (XFAIL)' if is_expected_to_fail else '')
+                util.print_test_result(tested, total, True, passed_string, test_path)
+        else:
+            passed_string = 'FAIL%s (%d)' % (' (XPASS)' if is_expected_to_fail else '', returncode)
+            util.print_test_result(tested, total, False, passed_string, test_path)
+            print("================================================")
+            print(stdout)
+            print("================================================")
+
+    return passed
+
+
+def run_snapshot_tests(args, tests):
+    execute_snapshot_cmd = get_platform_cmd_prefix()
+    generate_snapshot_cmd = get_platform_cmd_prefix()
+    if args.runtime:
+        execute_snapshot_cmd.append(args.runtime)
+        generate_snapshot_cmd.append(args.runtime)
+
+    execute_snapshot_cmd.extend([args.engine, '--exec-snapshot', 'js.snapshot'])
+
+    if sys.platform == 'win32':
+        generate_snapshot_cmd.append(args.engine[:-4] + '-snapshot.exe')
```

Comment:
Why is the `[:-4]` required?

Reply:
Because the name of the binary is different on Windows and Linux (jerry.exe and jerry). But you're right, we can do a platform independent solution with os.path.splitext().